src/platform/forms-system/src/js/fields/ObjectField.jsx

```diff
@@ -238,16 +236,9 @@ class ObjectField extends React.Component {
     );
 
     // Id's are not always unique on the review and submit page
-    const id =
-      isRoot && formContext.onReviewPage
-        ? Object.keys(idSchema)
-            .reduce((ids, key) => {
-              ids.push(idSchema[key].$id || '');
-              return ids;
-            }, [])
-            .filter(k => k)
-            .join('_')
-        : idSchema.$id;
+    const id = `${idSchema.$id}${
+      typeof pageIndex === 'undefined' ? '' : `_${pageIndex}`
+    }`;
```

Comment:
In [a previous PR](https://github.com/department-of-veterans-affairs/vets-website/pull/12449/files#diff-b939f1575c6f2b2bd2da83144e658e934e437461482528fe4c035b4252787fcf), I thought combining all the IDs would solve this problem, but it didn't include the page index. In this update, the page index is appended - the unit test is updated to use the same `idSchema` but provide different `formContext.pagePerItemIndex` properly.

Reply:
Well apparently the e2e test for Education form 10203 failed because of this change, so I've reverted & modified it to include the page index.